receiver/dockerreceiver/config.go

```diff
@@ -0,0 +1,61 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package dockerreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/dockerreceiver"
+
+import (
+	"errors"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/receiver/scraperhelper"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/docker"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/dockerreceiver/internal/metadata"
+)
+
+var _ component.Config = (*Config)(nil)
+
+type Config struct {
```

Comment:
The approach I had in mind was for the common code for `docker_stats` receiver and the new `docker` receiver to not be duplicated, but instead extracted to a common package that would be used by both receivers. This is what we did for the Logging and Debug exporter here: https://github.com/open-telemetry/opentelemetry-collector/tree/v0.96.0/exporter/internal/common/. 

I understand that this is not as straightforward if the new `docker` receiver accepts configuration in different structure, but I think it's still doable (or maybe we can go with no changes to the structure of the config, as proposed in the [comment above](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/32023/files#r1547655679)).

Reply:
There already exists an `internal/docker` package, is it a good place for it or do we want to keep it in a separate package?

Reply:
The `internal/docker` package defines the Docker client for use in other collector packages. I think having one package doing one thing makes it easier to work with a package.

The code we want to extract is concerned with code common to both Docker and Docker Stats receiver. How about package `internal/docker/receiver` or `receiver/internal/docker`?

Reply:
@andrzej-stencel after initial tries, I'm reluctant to this idea

1. It seems that even in case of simple components like `debug` and `logging` exporter code duplication is not easy to avoid: https://github.com/open-telemetry/opentelemetry-collector/commit/9731ea3281de3c14141c8ccd7fab0d14ee9aac73
2. In case of this (these) receiver(s), the subcomponent `metricsReceiver` has `Config` embedded inside. To avoid recursive imports, we would have to move `Config` to the common package, which makes little sense because we already want to change its specification. Currently, metric receiver's code is the most crucial part of the receiver, so it makes little sense to create a common package without it.

To sum up, it seems that the only advantage of moving this this way is making the reviewing process earlier. If that's what you wanted to achieve, then I can agree and do it. Otherwise I'd avoid creating a common subpackage.